### ISSUE TITLE
hook-zmq can't find sys/un.h while building

### ISSUE THREAD

**Issue by AlecJourdain** [pyinstaller/pyinstaller#4813]
## Description of the issue

I build an application using pyinstaller, everything work correctly but when I'm putting --log-level=DEBUG I noticed that something is missing but it didn't failed the build. 

On french computer the build failed with utf-8 codec 0xff (caused by the accent) but on english the build is completed correctly.

I know that this header is needed for linux build, but for windows it's not needed to build my app.

## Context information (for bug reports)

* Output of `pyinstaller --version`: ```3.5```
* Version of Python: e.g. 3.7.6
* Platform:  Windows 10 (english settings)
* Did you also try this on another platform? Does it work there?

As long as the computer is in english the build can be run. There is still error message when running pyinstaller with debug

### Stacktrace / full error message

```
15939 INFO: Loading module hook "hook-zmq.py"...
15940 DEBUG: Collecting submodules for zmq.backend
18092 DEBUG: collect_submodules - Found submodules: {'zmq.backend.cython', 'include', 'zmq.backend.cython.context', "'sys/un.h':", 'zmq.backend.cython.message', 'open', 'zmq.backend', 'error', 'directory', 'c:\\dev\\project\\code\\src\\fw\\venv\\lib\\site-packages\\zmq\\backend\\cffi\\__pycache__\\_cffi_ext.c(266):', 'zmq.backend.select', 'zmq.backend.cffi', 'zmq.backend.cython._proxy_steerable', 'zmq.backend.cython.error', '_cffi_ext.c', 'zmq.backend.cython._poll', 'fatal', 'No', 'zmq.backend.cython.utils', 'or', 'zmq.backend.cython.constants', 'zmq.backend.cython._device', 'zmq.backend.cython.socket', 'such', 'file', 'zmq.backend.cython._version', 'Cannot', 'C1083:', 'file:'}
18162 WARNING: Hidden import "'sys/un.h':" not found!
18164 WARNING: Hidden import "open" not found!
18165 WARNING: Hidden import "error" not found!
18166 WARNING: Hidden import "directory" not found!
18167 WARNING: Hidden import "c:\dev\program\code\src\fw\venv\lib\site-packages\zmq\backend\cffi\__pycache__\_cffi_ext.c(266):" not found!
18216 WARNING: Hidden import "_cffi_ext.c" not found!
18218 WARNING: Hidden import "fatal" not found!
18219 WARNING: Hidden import "No" not found!
18220 WARNING: Hidden import "or" not found!
18221 WARNING: Hidden import "such" not found!
18223 WARNING: Hidden import "file" not found!
18224 WARNING: Hidden import "Cannot" not found!
18225 WARNING: Hidden import "C1083:" not found!
18227 WARNING: Hidden import "file:" not found!
```

**Comment by Legorooj** [comment]
Odd - it should run on all platforms, unless ZMQ uses a system codec?

**Comment by AlecJourdain** [comment]
I've set a new Windows 10 VM in french to be able to give you the output message when it failed

18439 INFO: Loading module hook "hook-zmq.py"...
----------------------------------------
'utf-8' codec can't decode byte 0xff in position 161: invalid start byte
These are the bytes around the offending byte:
----------------------------------------
Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Dev\Magnum\src\FW\venv\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\__main__.py", line 111, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\__main__.py", line 63, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\build_main.py", line 844, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\build_main.py", line 791, in build
    exec(code, spec_namespace)
  File "C:\Dev\Magnum\src\FW\integration/magnum.spec", line 20, in <module>
    noarchive=False)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\build_main.py", line 243, in __init__
    self.__postinit__()
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\datastruct.py", line 158, in __postinit__
    self.assemble()
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\build_main.py", line 502, in assemble
    module_hook.post_graph()
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\imphook.py", line 410, in post_graph
    self._load_hook_module()
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\building\imphook.py", line 377, in _load_hook_module
    self.hook_module_name, self.hook_filename)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\compat.py", line 793, in importlib_load_source
    return mod_loader.load_module()
  File "<frozen importlib._bootstrap_external>", line 407, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 907, in load_module
  File "<frozen importlib._bootstrap_external>", line 732, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\hooks\hook-zmq.py", line 18, in <module>
    hiddenimports = ['zmq.utils.garbage'] + collect_submodules('zmq.backend')
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 624, in collect_submodules
    repr(pkg_dir), package))
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 83, in exec_statement
    return __exec_python_cmd(cmd)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 72, in __exec_python_cmd
    txt = exec_python(*cmd, env=pp_env)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\compat.py", line 637, in exec_python
    return exec_command(*cmdargs, **kwargs)
  File "c:\dev\magnum\src\fw\venv\lib\site-packages\PyInstaller\compat.py", line 427, in exec_command
    out = out.decode(encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 161: invalid start byte

**Comment by AlecJourdain** [comment]
In this post I've seen a way to fix it, but I think that is not very clean to ignore errors. 

https://stackoverflow.com/questions/47692960/error-when-using-pyinstaller-unicodedecodeerror-utf-8-codec-cant-decode-byt

**Comment by Legorooj** [comment]
Ok I'll look into improving compat... Thanks for pointing this out.

**Comment by htgoebel** [comment]
Beside any encoding issue: `collect_submodules` return a list of wrong strings:
>  collect_submodules - Found submodules: {… 'include', … sys/un.h':, … 'open', … 'error', 'directory', 'c:\\dev\\project\\code\\src\\fw\\venv\\lib\\site-packages\\zmq\\backend\\cffi\\__pycache__\\_cffi_ext.c(266):', '…', 'fatal', 'No', 'zmq.backend.cython.utils', 'or', 'zmq.backend.cython.constants', 'zmq.backend.cython._device', 'zmq.backend.cython.socket', 'such', 'file', 'zmq.backend.cython._version', 'Cannot', 'C1083:', 'file:'}

Why does it pick up sys/un.h? Where does the error message cause from?

@AlecJourdain What exaclty does "English settings" and "french settings" mean? Which encoding does Python have, what is the filesystem-encoding? What is the string which fails to print (Use python -m pdb pyinstaller.py …` to get a debugger at the failing place)

**Comment by mzy2240** [comment]
I am having the same problem with pyinstaller 3.6 and pyzmq version 18.1.1 as well as 19.0.0. I did not use the hook-zmq file and I did find the `libzmq.cp36-win_amd64.pyd` in the cython subdirectory. Any help or comments are highly appreciated!
![image](https://user-images.githubusercontent.com/18620336/80179244-ba5e1180-85c5-11ea-834a-8f3761b5de4a.png)

**Comment by Legorooj** [comment]
@mzy2240 sigh sigh sigh. Yet ANOTHER thing to add to my todo-list: rewrite all of our encoding interfaces. Could you tell me what the value of the following are?

* `sys.stdout.encoding`
* `sys.getfilesystemencoding()`

**Comment by mzy2240** [comment]
![image](https://user-images.githubusercontent.com/18620336/80180234-e7abbf00-85c7-11ea-9bce-ec265fcac27c.png)
Wow really appreciate your quick reply!

**Comment by Legorooj** [comment]
@mzy2240 we just happen to be awake at the same time. And what language is the default on your laptop?

**Comment by mzy2240** [comment]
I just tried the possible solution mentioned by @AlecJourdain. However, after rerun the pyinstaller command, the first error becomes `_cffi_ext.c` not found. Please see attached:
![image](https://user-images.githubusercontent.com/18620336/80180425-5852db80-85c8-11ea-8cc9-480794e651ce.png)

**Comment by mzy2240** [comment]
> @mzy2240 we just happen to be awake at the same time. And what language is the default on your laptop?

English(US)
![image](https://user-images.githubusercontent.com/18620336/80180569-a9fb6600-85c8-11ea-88f8-99ef26eca9a1.png)

**Comment by Legorooj** [comment]
@mzy2240 I'll have to get back to you. Unicode handling definitely needs a rewrite. (Yay)

**Comment by mzy2240** [comment]
Do you think that missing of ffi-related file is due to the current unicode handling approach?

**Comment by Legorooj** [comment]
@mzy2240 maybe, but I don't know. It's possible, though unlikely. But when it comes to the depths of a codebase that hasn't finished dropping support for python 2, anything it possible.

**Comment by AlecJourdain** [comment]
@htgoebel  I'm trying to add python -m pbd, but I'm always getting the error about no module named google. 

When the computer is in french he only difference is the build failed and in english the build is completed. I can paste the log in french here (Hidden import and submodule have some word in french): 

26401 DEBUG: Collecting submodules for zmq.backend
30968 DEBUG: collect_submodules - Found submodules: {'No', 'zmq.backend.cffi', 'zmq.backend.cython._device', 'c:\\dev\\magnum\\src\\fw\\venv\\lib\\site-packages\\zmq\\backend\\cffi\\__pycache__\\_cffi_ext.c(266):', "d'ouvrir", 'zmq.backend.cython.constants', 'zmq.backend.cython.socket', 'zmq.backend.cython._version', 'such', 'zmq.backend.cython.message', 'zmq.backend.cython.utils', 'include:', 'or', 'zmq.backend.cython._poll', 'zmq.backend.cython', "'sys/un.h':", 'file', 'C1083:', 'le', 'zmq.backend.cython.context', '_cffi_ext.c', 'fatal', 'error', 'Impossible', 'zmq.backend.cython.error', 'fichier', 'zmq.backend.select', 'zmq.backend', 'directory'}
31028 WARNING: Hidden import "No" not found!
31142 WARNING: Hidden import "c:\dev\project\src\fw\venv\lib\site-packages\zmq\backend\cffi\__pycache__\_cffi_ext.c(266):" not found!
31145 WARNING: Hidden import "d'ouvrir" not found!
31152 WARNING: Hidden import "such" not found!
31155 WARNING: Hidden import "include:" not found!
31158 WARNING: Hidden import "or" not found!
31173 WARNING: Hidden import "'sys/un.h':" not found!
31176 WARNING: Hidden import "file" not found!
31186 WARNING: Hidden import "C1083:" not found!
31191 WARNING: Hidden import "le" not found!
31194 WARNING: Hidden import "_cffi_ext.c" not found!
31201 WARNING: Hidden import "fatal" not found!
31206 WARNING: Hidden import "error" not found!
31213 WARNING: Hidden import "Impossible" not found!
31216 WARNING: Hidden import "fichier" not found!
31220 WARNING: Hidden import "directory" not found!

**Comment by Legorooj** [comment]
@AlecJourdain what's the output in a English VM? The same section.

**Comment by AlecJourdain** [comment]
@Legorooj 

15939 INFO: Loading module hook "hook-zmq.py"...
15940 DEBUG: Collecting submodules for zmq.backend
18092 DEBUG: collect_submodules - Found submodules: {'zmq.backend.cython', 'include', 'zmq.backend.cython.context', "'sys/un.h':", 'zmq.backend.cython.message', 'open', 'zmq.backend', 'error', 'directory', 'c:\\dev\\project\\code\\src\\fw\\venv\\lib\\site-packages\\zmq\\backend\\cffi\\__pycache__\\_cffi_ext.c(266):', 'zmq.backend.select', 'zmq.backend.cffi', 'zmq.backend.cython._proxy_steerable', 'zmq.backend.cython.error', '_cffi_ext.c', 'zmq.backend.cython._poll', 'fatal', 'No', 'zmq.backend.cython.utils', 'or', 'zmq.backend.cython.constants', 'zmq.backend.cython._device', 'zmq.backend.cython.socket', 'such', 'file', 'zmq.backend.cython._version', 'Cannot', 'C1083:', 'file:'}
18162 WARNING: Hidden import "'sys/un.h':" not found!
18164 WARNING: Hidden import "open" not found!
18165 WARNING: Hidden import "error" not found!
18166 WARNING: Hidden import "directory" not found!
18167 WARNING: Hidden import "c:\dev\program\code\src\fw\venv\lib\site-packages\zmq\backend\cffi\__pycache__\_cffi_ext.c(266):" not found!
18216 WARNING: Hidden import "_cffi_ext.c" not found!
18218 WARNING: Hidden import "fatal" not found!
18219 WARNING: Hidden import "No" not found!
18220 WARNING: Hidden import "or" not found!
18221 WARNING: Hidden import "such" not found!
18223 WARNING: Hidden import "file" not found!
18224 WARNING: Hidden import "Cannot" not found!
18225 WARNING: Hidden import "C1083:" not found!
18227 WARNING: Hidden import "file:" not found!

**Comment by AlecJourdain** [comment]
@Legorooj  For having the output in french I need to modify compat.py like this : 
https://stackoverflow.com/questions/47692960/error-when-using-pyinstaller-unicodedecodeerror-utf-8-codec-cant-decode-byt

Because when I do not modify compat.py I'll have the UnicodeDecodeError: 'UTF-8 codec can't decode byte 0xff in position 161: invalid start byte. 

the reason I'm thinking about a language problem is that the only difference between the two build is the OS language.

**Comment by Legorooj** [comment]
@AlecJourdain on the *french* machine, what is the value of `sys.stdout.encoding`? And `sys.getfilesystemencoding()`

**Comment by mzy2240** [comment]
Creating a new conda environment with only pyinstaller and zmq solves the problem. The solution is mentioned here: https://github.com/zeromq/pyzmq/issues/852

**Comment by Legorooj** [comment]
@mzy2240 can I have `pip freeze` (or conda equivalent) of both the new and the old environments?

**Comment by mzy2240** [comment]
@Legorooj 
The old one:
`absl-py==0.9.0
alabaster==0.7.12
altgraph==0.17
anaconda-client==1.7.2
anaconda-project==0.8.3
appdirs==1.4.3
argh==0.26.2
asn1crypto==1.3.0
astor==0.8.0
astroid==2.3.3
astropy==4.0
asttokens==1.1.13
atomicwrites==1.3.0
attrs==19.3.0
autopep8==1.4.4
Babel==2.8.0
backcall==0.1.0
backports.os==0.1.1
backports.shutil-get-terminal-size==1.0.0
bcrypt==3.1.7
beautifulsoup4==4.8.2
birdseye==0.8.2
bitarray==1.2.1
bkcharts==0.2
bleach==3.1.0
bokeh==1.3.4
boto==2.49.0
Bottleneck==1.3.2
cached-property==1.5.1
Cartopy==0.17.0
certifi==2019.11.28
cffi==1.14.0
chardet==3.0.4
cheap-repr==0.3.1
click==7.1.1
cloudpickle==1.3.0
clyent==1.2.2
colorama==0.4.3
comtypes==1.1.7
conda==4.8.3
conda-package-handling==1.6.0
configparser==3.7.4
contextlib2==0.6.0.post1
coverage==5.1
crypto==1.4.1
cryptography==2.8
cupy==6.2.0
cycler==0.10.0
Cython==0.29.12
cytoolz==0.10.1
dask==2.13.0
decorator==4.4.1
defusedxml==0.6.0
diff-match-patch==20181111
distributed==2.13.0
docutils==0.16
dtaidistance==1.2.2
dukpy==0.2.2
easydict==1.9
elasticsearch==7.5.1
entrypoints==0.3
enum-compat==0.0.2
esa==1.0.1
et-xmlfile==1.0.1
fastcache==1.1.0
fastdtw==0.2.0
fastrlock==0.4
fastscript==0.1.4
flake8==3.7.9
Flask==1.1.1
Flask-Humanize==0.3.0
fsspec==0.7.1
future==0.18.2
gast==0.3.3
gdbn==0.1
gevent==1.4.0
gnumpy==0.2
graphviz==0.5.1
greenlet==0.4.15
grpcio==1.27.2
h5py==2.10.0
HeapDict==1.0.1
html5lib==1.0.1
humanize==0.5.1
hypothesis==5.5.4
idna==2.8
idstools==0.6.3
ifaddr==0.1.6
imageio==2.8.0
imagesize==1.2.0
imbalanced-learn==0.6.2
importlib-metadata==1.5.0
intervaltree==3.0.2
ipykernel==5.1.4
ipython==7.12.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.21
itsdangerous==1.1.0
javascripthon==0.10
jdcal==1.4.1
jedi==0.15.2
Jinja2==2.11.1
joblib==0.14.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==5.3.4
jupyter-console==6.1.0
jupyter-contrib-core==0.3.3
jupyter-contrib-nbextensions==0.5.1
jupyter-core==4.6.1
jupyter-echarts-pypkg==0.1.2
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.4
jupyter-nbextensions-configurator==0.4.1
jupyterlab==0.35.5
jupyterlab-server==0.2.0
jupyterthemes==0.20.0
Keras==2.2.4
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
keyring==21.1.1
kiwisolver==1.1.0
kymatio==0.1.0
Lasagne==0.1
lazy-object-proxy==1.4.3
lesscpy==0.13.0
littleutils==0.2.1
llvmlite==0.29.0
lml==0.0.2
locket==0.2.0
lxml==4.5.0
macropy3==1.1.0b2
Mako==1.0.14
Markdown==3.1.1
MarkupSafe==1.1.1
matplotlib==3.1.3
mccabe==0.6.1
mebo==0.1.0b1
menuinst==1.4.16
mistune==0.8.4
mkl-fft==1.0.15
mkl-random==1.1.0
mkl-service==2.3.0
mock==4.0.1
more-itertools==8.2.0
mpmath==1.1.0
msgpack==1.0.0
msgpack-numpy==0.4.4.3
msgpack-python==0.5.6
multipledispatch==0.6.0
Naked==0.1.31
nbconvert==5.6.1
nbdev==0.2.9
nbformat==5.0.4
networkx==2.4
neupy==0.8.2
nltk==3.4.5
nolearn==0.6.0
nose==1.3.7
notebook==6.0.3
numba==0.45.1
numexpr==2.7.1
numpy==1.18.2
numpydoc==0.9.2
oauthlib==3.1.0
olefile==0.46
openpyxl==3.0.3
outdated==0.2.0
OWSLib==0.18.0
packaging==20.1
paho-mqtt==1.4.0
-e git+https://github.com/e2nIEE/pandapower.git@908e8bf55d844b044a3bd286d49ac627e3f5cff6#egg=pandapower
pandas==1.0.3
pandocfilters==1.4.2
paramiko==2.7.1
parso==0.5.2
partd==1.1.0
path==13.1.0
pathlib2==2.3.5
pathtools==0.1.2
patsy==0.5.1
pefile==2019.4.18
pep8==1.7.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.0.0
pipenv==2018.11.26
pkginfo==1.5.0.1
pluggy==0.13.1
ply==3.11
profilehooks==1.11.2
progressbar2==3.34.3
prometheus-client==0.7.1
prompt-toolkit==3.0.3
protobuf==3.11.4
psutil==5.7.0
py==1.8.1
pybind11==2.3.0
pycodestyle==2.5.0
pycosat==0.6.3
pycparser==2.19
pycrypto==2.6.1
pycuda==2019.1.1
pycurl==7.43.0.5
pydnp3==0.1.0
pydocstyle==4.0.1
pyecharts==0.5.11
pyecharts-javascripthon==0.0.6
pyecharts-jupyter-installer==0.0.3
pyepsg==0.4.0
pyflakes==2.1.1
pyforest==1.0.2
Pygments==2.5.2
PyInstaller==3.6
pykdtree==1.3.1
pylint==2.4.4
PyNaCl==1.3.0
pyodbc===4.0.0-unsupported
Pyomo==5.6.6
pyOpenSSL==19.1.0
pyparsing==2.4.6
pypiwin32==223
PYPOWER==5.1.4
pyproj==1.9.6
pypsa==0.14.1
pyreadline==2.1
pyrsistent==0.15.7
pyshark==0.4.2.9
pyshp==2.1.0
PySocks==1.7.1
pytemperature==1.0
pytest==5.4.1
pytest-arraydiff==0.3
pytest-astropy==0.8.0
pytest-astropy-header==0.1.2
pytest-doctestplus==0.5.0
pytest-openfiles==0.4.0
pytest-remotedata==0.3.2
python-dateutil==2.8.1
python-jsonrpc-server==0.3.4
python-language-server==0.31.9
python-utils==2.3.0
pytools==2019.1.1
pytz==2019.3
PyUtilib==5.7.1
PyWavelets==1.1.1
pywin32==227
pywin32-ctypes==0.2.0
pywinpty==0.5.7
PyYAML==5.3
pyzmq==18.1.1
QDarkStyle==2.8
QtAwesome==0.7.0
qtconsole==4.6.0
QtPy==1.9.0
readme-renderer==24.0
requests==2.22.0
requests-file==1.4.3
requests-oauthlib==1.3.0
requests-toolbelt==0.9.1
rise==0.0.0
rope==0.16.0
Rtree==0.9.3
ruamel-yaml==0.15.87
scikit-cuda==0.5.3
scikit-image==0.16.2
scikit-learn==0.22.1
scikit-umfpack==0.3.2
scikit-video==1.1.11
scipy==1.4.1
seaborn==0.10.0
selenium==3.141.0
selenium-requests==1.3
Send2Trash==1.5.0
Shapely==1.6.4.post2
shellescape==3.4.1
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.14.0
sklearn==0.0
snowballstemmer==2.0.0
sortedcollections==1.1.2
sortedcontainers==2.1.0
soupsieve==2.0
Sphinx==2.4.4
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
sphinxcontrib-websupport==1.2.1
spyder==4.1.1
spyder-kernels==1.9.0
SQLAlchemy==1.3.15
ss==0.4.8
statsmodels==0.11.0
sympy==1.5.1
tables==3.6.1
tabulate==0.8.3
tblib==1.6.0
tensorboard==1.13.1
tensorflow==1.13.1
tensorflow-estimator==1.13.0
termcolor==1.1.0
terminado==0.8.3
testpath==0.4.4
Theano==1.0.4
tldextract==2.2.1
toolz==0.10.0
torch==1.1.0
torchvision==0.3.0
tornado==6.0.3
tqdm==4.31.1
traitlets==4.3.3
tweepy==3.8.0
twine==1.13.0
typed-ast==1.4.1
ujson==1.35
unicodecsv==0.14.1
urllib3==1.25.8
versioneer==0.18
virtualenv==16.4.3
virtualenv-clone==0.5.2
watchdog==0.10.2
wcwidth==0.1.8
webencodings==0.5.1
Werkzeug==1.0.0
widgetsnbextension==3.5.1
win-inet-pton==1.1.0
win-unicode-console==0.5
wincertstore==0.2
wrapt==1.12.1
xlrd==1.2.0
XlsxWriter==1.2.8
xlwings==0.18.0
xlwt==1.3.0
yapf==0.28.0
yeelight==0.4.4
yellowbrick==1.1
zeroconf==0.21.3
zerorpc==0.6.3
zict==2.0.0
zipp==2.2.0`

The new one (which is working):
altgraph==0.17
certifi==2020.4.5.1
future==0.18.2
msgpack==1.0.0
pefile==2019.4.18
PyInstaller==3.6
pywin32-ctypes==0.2.0
pyzmq==19.0.0
wincertstore==0.2
zmq==0.0.0

**Comment by Legorooj** [comment]
@mzy2240 thanks. Well if it's working, then we only need to rewrite the encodings.

**Comment by AlecJourdain** [comment]
@Legorooj 
C:\Dev\Project\src\FW\venv\Scripts>python
Python 3.7.6 (tags/v3.7.6:43364a7ae0, Dec 19 2019, 00:42:30) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> sys.stout.encoding
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'sys' is not defined
>>> import sys
>>> sys.stout.encoding
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'sys' has no attribute 'stout'
>>> sys.stdout.encoding
'utf-8'
>>> sys.getfilesystemencoding()
'utf-8'
>>>

@mzy2240 So the solution is to setup a new venv with most recent version of pyinstaller and pyzmq ?

**Comment by mzy2240** [comment]
@AlecJourdain Yes

**Comment by Legorooj** [comment]
@AlecJourdain if that doesn't work know that I have 1 PR to finish, one website, then I'll be onto this. (ETA: 1 week until the PR I submit is ready for review)

**Comment by ethantenison** [comment]
Hello!

I'm having this exact same issue. The language of my computer is English, but the app created through pyinstaller uses Arabic text. I created a virtual environment and updated all the packages, but it still doesn't work. =(

**Comment by rokm** [comment]
The originally reported issue (parts of error message from attempted compilation of C extension module for `zmq.backend.cffi`) should be fixed by #5244. 

The UnicodeDecodeError issue reported later occurs due to localized error messages by Visual C compiler when non-English locale is used. This is now effectively worked around by pyinstaller/pyinstaller-hooks-contrib#59 (available from `pyinstaller-hooks-contrib` v.2020.10 on). As this work-around prevents the attempt at compiling the C extension module for the `cffi` backend, it also prevents the originally reported issue.

With this, the issues with the `zmq` hook should be taken care of. If not, please open a new issue with updated details.
